x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialUtils.java

```diff
@@ -33,7 +33,8 @@ public static Polygon createRegularGeoShapePolygon(Circle circle, int gons) {
         result[0] = new double[gons+1];
         result[1] = new double[gons+1];
         for(int i=0; i<gons; i++) {
-            double angle = i * (360.0 / gons);
+            // make sure we do not start at angle 0 or we have issues at the poles
+            double angle = i * (360.0 / gons) + 5.0;
```

Comment:
That fails for gons == 72. It is a bit tricky because it not only fails at 0 but also if it get pretty close to 180. I think adding something like `(360.0 / gons) / 3` might work. Basically we need to add anything that is guaranteed never to be equal to `(360.0 / gons)` or `(360.0 / gons) / 2` for any value of gons otherwise we hit either 0 or 180.

Reply:
Yes, I did realise of that as well. I have another thought and still this seems wrong as the final polygon makes no sense. We cannot really use this algorithm to create a polygon that contains a pole so I am thinking in throwing an error in that case for the time being. 

We can think later on if we can improve the algorithm for such case. It is difficult to solve a spherical problem using planar logic.